src/org/labkey/test/tests/TextChoiceSampleTypeTest.java

```diff
@@ -0,0 +1,653 @@
+        final String textField = "txtField";
+        log(String.format("Add a text field '%s' that will be converted to a TextChoice field.", textField));
+        sampleTypeDefinition.addField(new FieldDefinition(textField, ColumnType.String));
+
+        sampleTypeHelper.createSampleType(sampleTypeDefinition);
+
+        log("Edit the sample type and validate that the expected values are shown. They should be in alphabetical order.");
+
+        UpdateSampleTypePage updatePage = sampleTypeHelper.goToEditSampleType(sampleTypeName);
+        DomainFieldRow fieldRow = updatePage.getFieldsPanel().getField(textChoiceFieldName);
+        fieldRow = fieldRow.expand();
+
+        List<String> actualValues = fieldRow.getTextChoiceValues();
+
+        // Sort only the original list, not the list returned from the UI.
+        Collections.sort(expectedValues);
+
+        checker().verifyEquals("Values for field not as expected.", expectedValues, actualValues);
+
+        log("Validate that none of the values are shown as locked.");
+        actualValues = fieldRow.getLockedTextChoiceValues();
+
+        checker().verifyTrue(String.format("Fields '%s' are locked, they should not be.", String.join(",", actualValues)),
+                actualValues.isEmpty());
+
+        checker().screenShotIfNewError("ST_Designer_Initial_Values_Not_Correct");
+
+        List<String> expectedConvertedValues = List.of("Apple", "Banana");
+        log("Add some samples to the sample type and set the TextChoice field for some of the samples.");
+        sampleTypeHelper.goToSampleType(sampleTypeName);
+        List<Map<String, String>> samples = new ArrayList<>();
+
+        samples.add(Map.of(textChoiceFieldName, valuesUsed.get(0), textField, ""));
+        samples.add(Map.of(textChoiceFieldName, "", textField, expectedConvertedValues.get(0)));
+        samples.add(Map.of(textChoiceFieldName, valuesUsed.get(1), textField, ""));
+        samples.add(Map.of(textChoiceFieldName, valuesUsed.get(1), textField, ""));
+        samples.add(Map.of(textChoiceFieldName, "", textField, expectedConvertedValues.get(1)));
+
+        sampleTypeHelper.bulkImport(samples);
+
+        log(String.format("Edit the sample type again and validate that TextChoice values '%s' are locked.", String.join(", ", valuesUsed)));
+
+        waitAndClickAndWait(Locator.lkButton("Edit Type"));
+        updatePage = new UpdateSampleTypePage(getDriver());
+
+        fieldRow = updatePage.getFieldsPanel().getField(textChoiceFieldName);
+        fieldRow = fieldRow.expand();
+
+        List<String> lockedValues = fieldRow.getLockedTextChoiceValues();
+
+        // Sort the result from the UI, at this point more concerned about the values and not the order.
+        Collections.sort(lockedValues);
+        Collections.sort(valuesUsed);
+
+        checker()
+                .withScreenshot("ST_Designer_Locked_Values_Error")
+                .verifyEquals("Locked values not as expected.", valuesUsed, lockedValues);
```

Comment:
Could use the new `assertEqualsSorted` method here (and below):
```suggestion

        checker()
                .withScreenshot("ST_Designer_Locked_Values_Error")
                .wrapAssertion(() -> LabKeyAssert.assertEqualsSorted("Locked values not as expected.", valuesUsed, lockedValues));
```

Might even be worth creating `DeferredErrorCollector.verifyEqualsSorted` instead of using `wrapAssertion`.